cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -593,6 +603,184 @@ private Future<List<Condition>> maybeRestartConnectorTask(Reconciliation reconci
+            conditions.add(StatusUtils.buildWarningCondition("ListOffsets", message));
+            return Future.succeededFuture(conditions);
+        }
+        String configMapName = connectorSpec.getListOffsets().getConfigMapReference().getName();
+        return apiClient.getConnectorOffsets(reconciliation, host, port, connectorName)
+                .compose(offsets -> {
+                    Map<String, String> data = new HashMap<>(1);
+                    data.put(getConnectorOffsetsConfigMapEntryKey(connectorName), offsets);
+                    return generateListOffsetsConfigMap(configMapName, resource, data);
+                })
+                .compose(configMap -> configMapOperations.reconcile(reconciliation, resource.getMetadata().getNamespace(), configMapName, configMap))
+                .compose(v -> removeConnectorOffsetsAnnotation(reconciliation, resource))
+                .compose(v -> Future.succeededFuture(conditions), throwable -> {
+                    // Don't fail reconciliation on error from listing offsets - add a warning and repeat list on next reconcile
+                    String message = "Encountered error listing connector offsets. " + throwable.getMessage();
+                    LOGGER.warnCr(reconciliation, message);
+                    conditions.add(StatusUtils.buildWarningCondition("ListOffsets", message));
+                    return Future.succeededFuture(conditions);
+                });
+    }
+
+
+
+    /**
+     * Generates a connector offsets ConfigMap with the provided data.
+     * If there is already an existing ConfigMap it includes the existing labels and data, and uses the existing owner references (if present).
+     *
+     * @return The generated ConfigMap
+     */
+    @SuppressWarnings({ "rawtypes" })
+    private Future<ConfigMap> generateListOffsetsConfigMap(String configMapName, CustomResource resource, Map<String, String> data) {
+        String configMapNamespace = resource.getMetadata().getNamespace();
+        return configMapOperations.getAsync(configMapNamespace, configMapName)
+                .compose(existingConfigMap -> {
+                    Labels labels = Labels.fromMap(resource.getMetadata().getLabels());
+                    List<OwnerReference> ownerReferences;
+                    if (existingConfigMap == null) {
+                        ownerReferences = List.of(ModelUtils.createOwnerReference(resource, false));
+                    } else {
+                        labels = labels.withAdditionalLabels(existingConfigMap.getMetadata().getLabels());
+                        List<OwnerReference> existingOwnerReferences = existingConfigMap.getMetadata().getOwnerReferences();
+                        ownerReferences = existingOwnerReferences.isEmpty() ? List.of(ModelUtils.createOwnerReference(resource, false)) : existingOwnerReferences;
+                        data.putAll(existingConfigMap.getData());
+                    }
```

Comment:
This is IMHO wrong. You should patch the resource and not cherry-pick some individual aspects of it such as the owner references.

Reply:
I've updated the logic to patch the resource as you suggested